### ISSUE TITLE
Shared rsync folders allways get root.root as user and group

### ISSUE THREAD

**Issue by eriksenonline** [mitchellh/vagrant#5398]
I realy do not know if this is a bug or not. I have tried a lot of different stuff and searched the net but I can not get this to work. I have tried both on windows with cygwin and on my Mac, same problem. 

No mather what I do the files on /var/www/html/current get root as user and root as group.

Here is a logfile with INFO setting for vagrant up  
https://gist.github.com/eriksenonline/dba4dcbdc1ba48a7ba07

In my vagrant file I have tried this:

```ruby
    config.vm.synced_folder "html/", "/var/www/html/current", 
    type: "rsync", 
    owner: "vagrant", group: "vagrant", 
    rsync__exclude: [".git/", "var/", "/media"], 
    rsync__args: ["--verbose", "--rsync-path='sudo rsync'", "--archive", "--delete", "-z"],
    rsync__auto: "true"
```

I have also tried without the rsync args I have tried to put owner and group in the rsync args but nothing works.

**Comment by sethvargo** [comment]
Hi @eriksenonline 

Some synced folders do not support modifying the user/group at this time. I believe rsync is one of those types.

**Comment by eriksenonline** [comment]
ok, thanks for the answer.
When I sync the file rights should match the user executing the command. I have tried with and without the "sudo rsync" and still the files get root.root But am I not running as the vagrant user when I execute the sync-auto or vagrant up with rsync? If so the files should get the vagrant.vagrant file rights?

**Comment by mitchellh** [comment]
This is fixed in master

**Comment by ladjack** [comment]
Hi,

I know this issue closed long time ago but I think I got pretty much the same problem now. I use macOS Sierra as host system, latest Vagrant app 1.8.6 and virtual box 5.1.8 installed and CentOS 7 as guest OS. Using rsync to sync files. 

I can rsync both regular files and symlinks (copy symlinks as symlinks, -l option) with no problems, but only regular files/folders preserve original owner & group. Symlinks always get root as owner and group on target system (centos). 

I did several things like providing "owner" and "group" options in vagrant file with "vagrant" user as value and also specified "--rsync-path='sudo rsync'" as well:

```ruby
config.vm.synced_folder "./data", "/vagrant",
type: "rsync",
rsync__exclude: ".git/",
owner: "vagrant", group: "vagrant", 
rsync__args: ["-v", "-a", "-z", "--delete", "--rsync-path='sudo rsync'"]
```

 Nothing helped. I even switched to another box "bento centos" - with no luck. Finally if I leave sync section in vagrant file as simple as this:

```ruby
config.vm.synced_folder "./data", "/vagrant",
type: "rsync",
rsync__exclude: ".git/",
rsync__args: ["-v", "-a", "-z", "--delete"]
```

still symlinks on target filesystem get root as owner & group. 

Is there anything I can do to find out the reason for such behavior?

Thanks,
Eugene
